src/types/index.ts

```diff
@@ -407,25 +276,116 @@ export interface Tag {
   id: string;
   isOfficial: boolean;
 }
-
 export interface ReplComment {
   id: number;
   body: string;
   user: User;
 }
 
-// Response types
+/*****************************************************************
+ * * Miscalleneous / React Types / Function Args
+ *****************************************************************/
 
-export type GraphResponse<T> = Promise<T | never>;
+export enum HandshakeStatus {
+  Ready = "ready",
+  Error = "error",
+  Loading = "loading",
+}
 
-export type ReplQueryOutput = GraphResponse<{
-  repl: Repl;
-}>;
+export type StrError = { error: string };
+export type NullableStrError = { error: string | null };
+export type VoidPromise = Promise<void>;
+export type VoidFunction = () => void;
```

Comment:
will these types show up in the documentation? I think it would be unusual, and i'd rather just see `Promise<void>` instead. Same with VoidFunction

Reply:
Alrighty, will revert this.

Reply:
StrError and NullableStrError don't seem to show up in the docs, so I'll keep those.